services/src/main/java/org/keycloak/protocol/oid4vc/issuance/mappers/OID4VCIssuedAtTimeClaimMapper.java

```
@@ -123,13 +124,22 @@ public void setClaimsForCredential(VerifiableCredential verifiableCredential,
                 .orElseGet(() -> Optional.ofNullable(verifiableCredential.getIssuanceDate())
                         .orElse(Instant.now()));
 
+        Instant normalizedIat = new TimeClaimNormalizer(userSessionModel.getRealm())
+                .normalize(iat, Instant.now());
+
         // truncate is possible. Return iat if not.
         Instant iatTrunc = Optional.ofNullable(mapperModel.getConfig())
                 .flatMap(config -> Optional.ofNullable(config.get(TRUNCATE_TO_TIME_UNIT_KEY)))
-                .filter(String::isEmpty)
+                .filter(val -> !val.isEmpty())
                 .map(ChronoUnit::valueOf)
-                .map(iat::truncatedTo)
-                .orElse(iat);
+                .map(unit -> {
+                    // Only support units that Instant.truncatedTo can handle reliably
+                    return switch (unit) {
+                        case MINUTES, HOURS, DAYS -> normalizedIat.truncatedTo(unit);
+                        default -> iat;
+                    };
+                })
```

Let the `truncatedTo` method handle this logic? It is also dangerous to simply ignore unsupported options, leaking `iat` without normalization.

---

> Let the `truncatedTo` method handle this logic? It is also dangerous to simply ignore unsupported options, leaking `iat` without normalization.

Good point. The logic now uses `truncatedTo` and catches `UnsupportedTemporalTypeException`. 
If a unit is unsupported, we log a warning and keep the normalized value without truncation.

---

I still find it dangerous not to fail. There is usually a method that can be implemented to validate config values, so we fail early if not MINUTES, HOURS, DAYS.